src/app/components/merge/MergeModal.js

```diff
@@ -1,236 +1,114 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import PropTypes from 'prop-types';
 import { c } from 'ttag';
-import { useApi, useEventManager, useModals, FormModal, ResetButton, PrimaryButton } from 'react-components';
+import { useEventManager, FormModal, ResetButton, PrimaryButton } from 'react-components';
 
-import { getContact, addContacts, deleteContacts } from 'proton-shared/lib/api/contacts';
-import { noop } from 'proton-shared/lib/helpers/function';
-import { prepareContact as decrypt } from '../../helpers/decrypt';
-import { prepareContact as encrypt } from '../../helpers/encrypt';
-import { merge } from '../../helpers/merge';
-import { moveInGroup } from '../../helpers/array';
-import { OVERWRITE, CATEGORIES, SUCCESS_IMPORT_CODE } from '../../constants';
-
-import ContactDetails from './ContactDetails';
-import MergeContactPreview from './MergeContactPreview';
 import MergeModalContent from './MergeModalContent';
-import MergeTable from './MergeTable';
 import MergingModalContent from './MergingModalContent';
-import { splitKeys } from 'proton-shared/lib/keys/keys';
-
-const { OVERWRITE_CONTACT } = OVERWRITE;
-const { IGNORE } = CATEGORIES;
 
-const MergeModal = ({ contacts, userKeysList, onMerge = noop, ...rest }) => {
-    const api = useApi();
+const MergeModal = ({ contacts, contactID, userKeysList, ...rest }) => {
     const { call } = useEventManager();
-    const { createModal } = useModals();
 
     const [isMerging, setIsMerging] = useState(false);
+    const [mergeFinished, setMergeFinished] = useState(false);
     const [model, setModel] = useState({
```

Comment:
```suggestion
    const [model, setModel] = useState(() => ({
```

to just having it evaluated once

Reply:
Why does React re-compute the value if it's not going to use it... ?¿